param_slice.m

```diff
@@ -22,16 +23,38 @@
 end
 %inds = varargin(2:2:end);
 
-full_ind = ones(numel(all_fields), 1);
-for fs_it = 1:n_fields
-    ind = get_field_ind(p_list, fields{fs_it});
-    if isempty(ind)
-        error('Specified field does not exist in parameter struct.');
+if n_fields == numel(all_fields)
+    full_ind = ones(numel(all_fields), 1);
+    for fs_it = 1:n_fields
+        ind = get_field_ind(p_list, fields{fs_it});
+        if isempty(ind)
+            error('Specified field does not exist in parameter struct.');
+        end
+        full_ind(ind) = inds(fs_it);
+    end
+    full_ind = num2cell(full_ind);
+    
+    p_it = sub2ind(siz, full_ind{:});
+else
```

Comment:
This works, but it doesn't seem very efficient to build a matrix containing the indices of every possible parameter combination and then pruning it.

How about iterating over the possible combinations of fields that have multiple matches? This would make it easy to implement a more general slice where you can specify an array of indices for each field. Right now you can only get one or all.